meilidb-http/src/routes/index.rs

```diff
@@ -12,104 +17,312 @@ use crate::models::token::ACL::*;
+    let updated_at = index
+        .main
+        .updated_at(&mut reader)
+        .map_err(ResponseError::internal)?
+        .ok_or(ResponseError::internal("Updated date not found"))?;
+
+    let response_body = IndexResponse {
+        name,
+        uid,
+        created_at,
+        updated_at,
+    };
 
-    match schema {
-        Some(schema) => {
-            let schema = SchemaBody::from(schema);
-            Ok(tide::response::json(schema))
-        }
-        None => Ok(
-            tide::response::json(json!({ "message": "missing index schema" }))
-                .with_status(StatusCode::NOT_FOUND)
-                .into_response(),
-        ),
-    }
+    Ok(tide::response::json(response_body))
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+struct IndexCreateRequest {
+    name: String,
+    schema: Option<SchemaBody>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+struct IndexCreateResponse {
+    name: String,
+    uid: String,
+    schema: Option<SchemaBody>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    update_id: Option<u64>,
+    created_at: DateTime<Utc>,
+    updated_at: DateTime<Utc>,
 }
 
 pub async fn create_index(mut ctx: Context<Data>) -> SResult<Response> {
     ctx.is_allowed(IndexesWrite)?;
 
-    let index_name = ctx.url_param("index")?;
+    let body = ctx
+        .body_json::<IndexCreateRequest>()
+        .await
+        .map_err(ResponseError::bad_request)?;
 
-    let body = ctx.body_bytes().await.map_err(ResponseError::bad_request)?;
-    let schema: Option<Schema> = if body.is_empty() {
-        None
-    } else {
-        serde_json::from_slice::<SchemaBody>(&body)
-            .map_err(ResponseError::bad_request)
-            .map(|s| Some(s.into()))?
-    };
+    let generated_uid = generate_uid();
 
     let db = &ctx.state().db;
 
-    let created_index = match db.create_index(&index_name) {
+    let created_index = match db.create_index(&generated_uid) {
         Ok(index) => index,
         Err(e) => return Err(ResponseError::create_index(e)),
     };
 
     let env = &db.env;
     let mut writer = env.write_txn().map_err(ResponseError::internal)?;
 
-    match schema {
-        Some(schema) => {
-            let update_id = created_index
-                .schema_update(&mut writer, schema.clone())
-                .map_err(ResponseError::internal)?;
+    created_index
+        .main
+        .put_name(&mut writer, &body.name)
+        .map_err(ResponseError::internal)?;
+    created_index
+        .main
+        .put_created_at(&mut writer)
+        .map_err(ResponseError::internal)?;
+    created_index
+        .main
+        .put_updated_at(&mut writer)
+        .map_err(ResponseError::internal)?;
 
-            writer.commit().map_err(ResponseError::internal)?;
+    let schema: Option<Schema> = body.schema.clone().map(|s| s.into());
```

Comment:
I would prefer you to use short functions like so, avoiding lots of pipes/parentheses everywhere :)

```suggestion
    let schema: Option<Schema> = body.schema.clone().map(Into::into);
```